### ISSUE TITLE
Suggestion: Extract decoders into separate crate that doesn't do playback

### ISSUE THREAD

**Issue by Boscop** [RustAudio/rodio#292]
It would be nice to have a unified API to read audio files of different formats without pulling in cpal (or any other playback crate), for use cases that require reading audio files but not playing them.

**Comment by est31** [comment]
Yeah that's what audrey is for. @mitchmindtree has suggested a while ago I think to move the decoding part out of rodio to use audrey instead.

**Comment by Boscop** [comment]
Yeah, it would be nice if there was one crate that supports all codecs that any rust crate supports. E.g. if audrey also supported mp3 (and aac).
